### PR TITLE
easyinstall: Default no. of compiler threads based on available RAM

### PR DESCRIPTION
**Pull request by rdmark**

### CODE REVIEW

File: easyinstall.sh [hunk 1]
```diff
@@ -49,7 +49,8 @@ echo -e $logo
 CONNECT_TYPE="FULLSPEC"
 # clang v11 is the latest distributed by Buster
 COMPILER="clang++-11"
-CORES=1
+MEM=$(grep MemTotal /proc/meminfo | awk '{print $2}')
+CORES=$(awk 'BEGIN { x = '$MEM'; y = 524288; print (x / y) }' | numfmt --round=up --format=%.0f)
 USER=$(whoami)
```
Comment (uweseimet) [inline easyinstall.sh:54]: There might by an error somewhere. When I execute the two new lines on my PC I get
```
numfmt: invalid suffix in input: ‘62.4754’
```
MEM is set correctly, to 32755124 in my case.

Comment (rdmark) [reply easyinstall.sh]: I realized the latter problem myself and pushed a fix for that.

For the former issue, I wonder if your version of `numfmt` behaves differently from mine?

Comment (uweseimet) [reply easyinstall.sh]: I tried the fixed PR also on two PCs running Ubuntu. I get a different error now. On my local PC it's
```
numfmt: invalid suffix in input: ‘62.4754’
```
On Ubuntu it is
```
numfmt: invalid suffix in input: ‘1.85975’
[: -gt: unexpected operator
```
and
```
numfmt: invalid suffix in input: ‘46.8516’
[: -gt: unexpected operator
```

Comment (uweseimet) [reply easyinstall.sh]: @rdmark Why not use a less complicated solution, e.g.
```
MEM=$(grep MemTotal /proc/meminfo | awk '{print $2}')
CORES=`expr $MEM / 524288`
```

Comment (rdmark) [reply easyinstall.sh]: Thanks for tracking down a more portable solution. I added a branch for a fallback to 1 core when >512mb of memory is available.

**Review by uweseimet — APPROVED** [review]

**Comment by uweseimet** [comment]
@rdmark Just noticed one more thing: This should be 519168, not 512000.

**Comment by rdmark** [comment]
@uweseimet FWIW, it seems like low-end RPis actually report less than 512MB MemTotal in /proc/meminfo, according to some random google search results.

F.e. for RPi Zero: 445032KB

https://inokara.hateblo.jp/entry/2016/02/08/004354

It would probably be good to benchmark MemTotal on some key RPi models and adjust the value later.

**Comment by uweseimet** [comment]
@rdmark Would this matter in our case? The current change always uses at least 1 core, and if less memory is reported than is actually available we would still be better off than before.
Also, this report is from 2016. It may be wrong or outdated.

**Comment by rdmark** [comment]
@uweseimet It would matter for f.e. the RPi 3B, which has 1GB of memory. This has been my primary development RPi for several years. It benefits hugely from using 2+ cores for compilation. But if MemTotal is less than 2x512MB then it would only get one core by default, defeating the goal of this change.

**Comment by uweseimet** [comment]
@rdmark Can you test this? If not and you have doubts we should discard the PR and the ticket, because it would not be resolvable if one assumes that wrong memory amounts are reported.

**Comment by rdmark** [comment]
I can test it in 2 months. My RPis are all on a long ocean journey right now. 

I think we should merge as-is for now. It's an improvement for high end RPis regardless.

My hypothesis is that ~400MB is a better number. In the past, I've used 3 cores with clang and 2 cores with gcc for compiling piscsi on my RPi3B+. But some benchmarking is preferable over anecdotal data.

**Comment by uweseimet** [comment]
@rdmark I'm fine with changing the value from 512 MB to 400 MB.

**Comment by rdmark** [comment]
Using 450MB as divisor now.

**Comment by uweseimet** [comment]
@rdmark Sounds good. I have already approved, please merge any time.

**Pull request merged**